binary/reflect.go

```diff
@@ -864,8 +865,9 @@ func writeReflectJSON(rv reflect.Value, rt reflect.Type, w io.Writer, n *int64,
 		length := rt.Len()
 		if elemRt.Kind() == reflect.Uint8 {
 			// Special case: Bytearray
-			bytearray := rv.Interface()
-			WriteTo([]byte(Fmt("\"%X\"", bytearray)), w, n, err)
```

Comment:
I didn't like it. But with the old code it seemed to be hex encoding the call to eg pubKey.String()